discord/colour.py

```diff
@@ -112,6 +118,13 @@ def from_hsv(cls, h, s, v):
         rgb = colorsys.hsv_to_rgb(h, s, v)
         return cls.from_rgb(*(int(x * 255) for x in rgb))
 
+    @classmethod
+    def from_hex(cls, hexa):
+        """Constructs a :class:`Colour` from hex format.
+        .. versionadded:: 1.5
+        """
+        return cls(int(hexa.strip("#"), base=16))
```

Comment:
This method doesn't really feel like it belongs? Some people will begin to expect other formats like `0xFFFFFF` or people will confuse passing in `int` and expecting it to work etc. There's a bit too much confusion that can arise from this method compared ot the value it brings, which is shaving off one line where you do `int(string[1:], base=16)`

Reply:
To me it seemed a bit weird to me that it didn't exist seen as the hex format is the standard used in the GUI colour picker, and there's a way to convert the object into a hex string. It was more of a consistency thing to fit in with that, plus the existing RGB and HSV methods if that makes sense, but I could understand the functionality not necessarily being needed.

The `to_hex()` method is just there as a counter to the `from_hex()`, as I feel that should be available if the `from_hex()` ended up being implemented.

Reply:
To me this doesn't make sense as a method because "from hex" is the default. Creating an instance of a colour takes in an integer value, and in python you can provide `0xFFFFFF`. This is hex, this is an integer, and is already supported as the only argument in the constructor. Due to this, I don't see a purpose for a class method that does the same thing, but from a string instead.

Reply:
I think, there's no use of `from_hex` and `to_hex`.